modules/statistics/php/charts.class.inc

```diff
@@ -257,18 +277,36 @@ class Charts extends \NDB_Page
      */
     private function _handleSiteLineData()
     {
-        $DB = \NDB_Factory::singleton()->database();
+        $DB          = \NDB_Factory::singleton()->database();
+        $currentUser = \NDB_Factory::singleton()->user();
+
+        if ($currentUser->hasPermission('access_all_profiles')) {
+            $list_of_sites = \Utility::getSiteList(true, false);
+        } else {
+            $list_of_sites = $currentUser->getSiteList(true, false);
+        }
 
-        $recruitmentData      = array();
+        $params            = array();
+        $centerIDs         = array();
+        $list_of_sites_ids = array_keys($list_of_sites);
+        foreach ($list_of_sites_ids as $key => $siteID) {
+            $params[]            = ":id$key";
```

Comment:
```suggestion
            $params[]            = "$key";
```
This seam like a typo or bad copy/paste.
Same for next line.

Reply:
Hello @PapillonMcGill ,

There is not typo or copy/paste error in the line 293. Please let me explain you what this code is doing: This line is doing a variable interpolation so each of the elements(**params**) in the `$params[]` array will take the shape '`:idkey`' for example: '`:id0`', '`:id1`', etc. As this will be passed to the `IN` clause in the `pselectOne` function (**line 305**) makes no sense to pass '0', '1', etc which will be the case if we go as you proposed changing `$params[] = ":id$key"` for: `$params[] = "$key"`.

I hope this helps you to understand. Please don't hesitate to ask me if you have any other doubt.

Best,
Rolando Acosta (CHU Sainte-Justine)

Reply:
@racostas I see!